### PR TITLE
Catch bad IBFT2 round change msgs

### PR DESCRIPTION
**Pull request by rain-on**
When validating a RoundChange message, its important that not only
are there sufficient Prepare msgs in the PreparedCertificate, but
also that each are from a different validator (i.e. cannot simply
count the number msgs, but must count distinct authors).

Signed-off-by: Trent Mohay <trent.mohay@consensys.net>

### CODE REVIEW

File: consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java [hunk 1]
```diff
@@ -96,10 +96,11 @@ private boolean validateConsistencyOfPrepareCertificateMessages(
       return false;
     }
 
-    if (certificate.getPreparePayloads().size() < minimumPrepareMessages) {
+    if (certificate.getPreparePayloads().stream().map(SignedData::getAuthor).distinct().count()
```
Comment (jframe) [inline consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java:99]: Do all the prepare authors also need to be unique? Or is it enough the we have minimumPrepareMessages. I guess I'm wondering if need a hasDuplicateAuthors check like in the RoundChangeCertificateValidator.

**Comment by saltiniroberto** [comment]
Fixes #1736

File: consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java [hunk 2]
```diff
@@ -113,6 +117,18 @@ private boolean validateConsistencyOfPrepareCertificateMessages(
     return true;
   }
 
+  private boolean hasDuplicateAuthors(
+      final Collection<SignedData<PreparePayload>> preparePayloads) {
+    final long distinctAuthorCount =
```
Comment (usmansaleem) [inline consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java:122]: can return early with `true` if preparePayloads list is empty?

**Review by usmansaleem — APPROVED** [review]
LGTM. Comment can be ignored if that list will never be empty.

File: consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java (hunk 2 above)
Comment (rain-on) [reply consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/validation/RoundChangePayloadValidator.java:122]: Probably could return early, however there's probably not much of a performance gain, and adds lines of code which would then need to be tested, so would rather not ... but can be convinced if there's significant benefits.

**Review by usmansaleem — APPROVED** [review]
LGTM

**Pull request merged**